src/Servers/Kestrel/Core/src/Internal/Http/Http1ChunkedEncodingMessageBody.cs

```diff
@@ -182,15 +189,15 @@ private async Task PumpAsync()
             }
             finally
             {
-                _requestBodyPipe.Writer.Complete(error);
+                await _requestBodyPipe.Writer.CompleteAsync(error);
             }
         }
 
-        protected override Task OnStopAsync()
+        protected override ValueTask OnStopAsync()
         {
             if (!_context.HasStartedConsumingRequestBody)
             {
-                return Task.CompletedTask;
+                return new ValueTask();
```

Comment:
Nit:

```suggestion
                return default;
```

This should be done everywhere `return new ValueTask()` is called.